mixer/pkg/cache/lruCache.go

```diff
@@ -255,21 +262,35 @@ func (c *lruCache) Get(key interface{}) (interface{}, bool) {
 
 func (c *lruCache) Remove(key interface{}) {
 	c.Lock()
-	c.remove(key)
+
+	if index, ok := c.lookup[key]; ok {
+		c.remove(index)
+	}
+
 	c.Unlock()
 }
 
-func (c *lruCache) remove(key interface{}) {
-	if index, ok := c.lookup[key]; ok {
-		delete(c.lookup, key)
-		c.unlinkEntry(index)
-		c.linkEntryAtTail(index)
-		c.entries[index].key = nil
-		c.entries[index].value = nil
-		c.entries[index].expiration = math.MaxInt64
+func (c *lruCache) RemoveAll() {
+	for i := 1; i < len(c.entries); i++ {
```

Comment:
Should we have a lock around this entire for loop ? Is there a chance that the entries size might shrink ?

Reply:
No, the entries slice cannot change size during the lifetime of the structure.